src/Refine/Refine.cpp

```diff
@@ -786,7 +840,55 @@ void Refine( const int lv, const UseLBFunc_t UseLBFunc )
 #        endif
 
 
-//       (c1.3.4.3) check minimum density and pressure/internal energy
+
+//       (c1.3.4.3) convert density/phase to real and imaginary parts if patches were refined from phase to wave level
+#        if ( MODEL == ELBDM && ELBDM_SCHEME == ELBDM_HYBRID )
+         if ( Pedigree->flag && !amr->use_wave_flag[lv] && amr->use_wave_flag[lv+1] ) {
+            real amp, phase, stub, Re, Im ;
+
+#           ifdef GAMER_DEBUG
+//          Check whether dB wavelength is resolved
+            for (int k=0; k<FSize_CC; k++) {
+            for (int j=0; j<FSize_CC; j++) {
+            for (int i=0; i<FSize_CC; i++) {
+                  int kk  =  k;
+                  int kkp = (kk + 1) < FSize_CC  ? kk + 1 : kk    ;
+                  int kkm = (kk - 1) < 0         ? kk     : kk - 1;
+                  int ii  =  i;
+                  int iip = (ii + 1) < FSize_CC  ? ii + 1 : ii    ;
+                  int iim = (ii - 1) < 0         ? ii     : ii - 1;
+                  int jj  =  j;
+                  int jjp = (jj + 1) < FSize_CC  ? jj + 1 : jj    ;
+                  int jjm = (jj - 1) < 0         ? jj     : jj - 1;
+
+                  //check whether dB wavelength is resolved within the newly converted patch
+                  real dPhase = MAX(MAX(MAX(MAX(MAX(
+                  FABS(Flu_FData[PHAS][kk ][jj ][iip] - Flu_FData[PHAS][kk ][jj ][ii ]),
+                  FABS(Flu_FData[PHAS][kk ][jj ][ii ] - Flu_FData[PHAS][kk ][jj ][iim])),
+                  FABS(Flu_FData[PHAS][kk ][jjp][ii ] - Flu_FData[PHAS][kk ][jj ][ii ])),
+                  FABS(Flu_FData[PHAS][kk ][jj ][ii ] - Flu_FData[PHAS][kk ][jjm][ii ])),
+                  FABS(Flu_FData[PHAS][kkp][jj ][ii ] - Flu_FData[PHAS][kk ][jj ][ii ])),
+                  FABS(Flu_FData[PHAS][kk ][jj ][ii ] - Flu_FData[PHAS][kkm][jj ][ii ]));
+
+                  if ( dPhase > M_PI ) {
+                     Aux_Message ( stderr, "WARNING: Phase jump = %d > PI when refining patch from fluid (lv %d) to wave (lv %d) scheme!", dPhase, lv, lv + 1);
+                  }
```

Comment:
@hyschive 
No, we cannot guarantee it will not happen. But how to handle the case that it still works at one time step and fails in the next? In this case, we would have to switch back to the fluid scheme and then convert the next level to the wave scheme instead. I have not implemented this feature which is why one should decide on a level beforehand. But of course, this could easily be implemented within the existing code.